### DISCUSSION TITLE
Manual import is stuck when using Symfony Messenger based import execution in `pimcore/data-importer`

### DISCUSSION THREAD

**Discussion by localheinz** [pimcore/pimcore#14140] [Q&A / Support]
We are using `pimcore/pimcore` and `pimcore/data-importer`.

We have a set up where

- we use `supervisor` with the following configuration to ensure that we have workers to process the `pimcore_core`,  `pimcore_data_import`, and `pimcore_maintenance` queues
  
  ```ini
  [program:maintenance]
  autorestart=true
  autostart=true
  command=bash -c 'sleep 3600 && exec php /var/www/html/bin/console pimcore:maintenance --async'
  environment=MESSENGER_CONSUMER_NAME=consumer-%(program_name)s-%(process_num)02d
  redirect_stderr=true
  stdout_logfile=/dev/fd/1
  stdout_logfile_maxbytes=0
  
  [program:messenger-consume]
  autorestart=true
  autostart=true
  command=php /var/www/html/bin/console messenger:consume pimcore_core pimcore_data_import pimcore_maintenance --memory-limit=250M --time-limit=3600
  environment=MESSENGER_CONSUMER_NAME=consumer-%(program_name)s-%(process_num)02d
  numprocs=1
  process_name=%(program_name)s_%(process_num)02d
  redirect_stderr=true
  startsecs=0
  stdout_logfile=/dev/fd/1
  stdout_logfile_maxbytes=0
  ```
- we use [Symfony Messenger Based Import Execution](https://github.com/pimcore/data-importer/blob/v1.7.3/doc/01_Installation.md#symfony-messenger-based), configured via `config/packages/pimcore-data-importer.php`

  ```php
  <?php
  
  declare(strict_types=1);
  
  use Symfony\Component\DependencyInjection;
  
  return static function (DependencyInjection\Loader\Configurator\ContainerConfigurator $containerConfigurator): void {
      $containerConfigurator->extension('pimcore_data_importer', [
          'messenger_queue_processing' => [
              'activated' => true,
          ],
      ]);
  };
  
  ```
- we use Redis as transport mechanism for messages
- we do not have any cronjobs set up

When we start the import by clicking the **Start** button in the **Manual Execution** section, as seen below, the import is stuck

![CleanShot 2023-01-25 at 14 22 41@2x](https://user-images.githubusercontent.com/605483/214574364-2f2af55c-8943-43b4-8274-6bd6425fa1eb.png)

The table `bundle_data_hub_data_importer_queue` contains 30 entries, as shown above, but apparently there is no process here that picks up items from the `bundle_data_hub_data_importer_queue` table and dispatches them to the queue for asynchronous processing.

What are we missing here?

The documentation is unclear whether we need to set up a cronjob at all.

**Comment by mcop1** [comment]
Hello @localheinz ,

per default the symfony messenger queue for data-importer is deactivated, you have to activate it.
See `Symfony Messenger Based` section here https://pimcore.com/docs/data-importer/current/Installation.html

**Reply by localheinz** [reply]
@mcop1 

Thank you for your reply, we already have configured [Symfony Messenger Based Execution](https://github.com/pimcore/data-importer/blob/v1.7.3/doc/01_Installation.md#symfony-messenger-based) - I have updated the description above.

**Reply by mcop1** [reply]
@localheinz Just tried it in my local environment and had no issues. Started the import manually, executed the `messenger:consume` command, worked. 
Could yo please try to add the `-vv` option to your command? Maybe the logs can help.

**Reply by localheinz** [reply]
@mcop1 

It's great to hear that you do not have any issues. Unfortunately, it does not work for us when following the documentation at https://github.com/pimcore/data-importer. I wish I understood what was required to set up Symfony-Messenger-based import execution. The documentation is not clear about it, otherwise I would not have opened this discussion here.

**Reply by localheinz** [reply]
@mcop1 

I would be happy to help update the documentation, but in order to do that, I would need to understand the tasks required to set up Symfony-Messenger-based import execution processing.

Can you help me with that?

**Reply by mcop1** [reply]
@localheinz 

Sorry, if I wasn´t clear enough. I didn´t do anything else than you did.

- Activated symfony messenger queue in data-importer configuration
- Created data-importer configuration with a dummy csv file
- Clicked the `Start` button (manual execution)
- Run `messenger:consume` command

That´s why I was asking for logs, to see if there may be another problem.

**Reply by localheinz** [reply]
@mcop1 

Not sure what the problem was, but it works now.

Thank you!

**Reply by CardSys** [reply]
Hi,
we had the same problem.
Then found this hint: https://pimcore.com/docs/data-importer/current/Troubleshooting_FAQ.html

After I deleted the row from the lock_keys table, the messenger-based import worked
